tools/mkpkg/update_binary-addons

```diff
@@ -68,10 +68,24 @@ fi
 
 . "${MY_DIR}/update_common_functions"
 
-git_clone https://github.com/xbmc/xbmc ${KODI_DIR} ${KODI_BRANCH}
+get_ghtoken
+
+if [ -z "${GITHUB_API_TOKEN}" ]; then
+  git_clone https://github.com/xbmc/xbmc ${KODI_DIR} ${KODI_BRANCH}
+else
+  # check if depends are available
+  command -v curl >/dev/null 2>&1 || die "please install curl"
+  command -v jq >/dev/null 2>&1 || die "please install jq"
+
+  curl -s -L -H "Authorization: token ${GITHUB_API_TOKEN}" \
+    https://api.github.com/repos/xbmc/xbmc/contents/cmake/addons/bootstrap/repositories | awk \
+    '/download_url/ { gsub("\"|,", "", $2); system("mkdir -p '${KODI_DIR}'; (cd '${KODI_DIR}'; curl -O "$2")"); }' \
+    2>/dev/null
```

Comment:
This doesn't take KODI_BRANCH into account and will always download from the default/master branch.

Also please download the temporary repository.txt files to a subdir in TMPDIR (eg `$TMPDIR/bootstrap-repos`) instead of KODI_DIR.

Then just add another variable (eg KODI_REPOS) pointing to that dir (or the dir in the cloned kodi repo in the if branch above) to use in the for loop below

Reply:
This should be doable by the parameter `?ref=${KODI_BRANCH}` , will take a look.